homeassistant/components/calendar/__init__.py

```diff
@@ -52,6 +52,14 @@ def get_date(date):
     return dt.as_local(dt.parse_datetime(date["dateTime"]))
 
 
+def convert_date(date):
+    """Convert Z date/time."""
+    date = re.sub(
+        r"T", " ", re.findall(r"\d{4}-\d{2}-\d{2}\D\d{2}:\d{2}:\d{2}", date)[0]
```

Comment:
What is the format of the date that comes in? Might be able to use the util for this:

https://github.com/home-assistant/core/blob/dev/homeassistant/util/dt.py

Reply:
2017-11-26T12:00:00.000Z

I've already attempted to use this. Resulting error is string indices must be integers.

Reply:
Then that's a bug you're running into . That method `parse_datetime` is set up to convert exactly those dates.